python_ta/checkers/unnecessary_indexing_checker.py

```diff
@@ -106,8 +129,12 @@ def _is_redundant(index_node: Union[astroid.AssignName, astroid.Name], for_node:
     in the for loop's body.
     """
     if isinstance(index_node, astroid.AssignName):
-        return index_node.lookup(index_node.name)[1][0] != for_node.target or \
-               not isinstance(index_node.parent, astroid.AugAssign)
+        if index_node.lookup(index_node.name)[1] != ():
```

Comment:
This is good, but you need a matching `else` branch to ensure that a boolean is always returned.